main.go

```diff
@@ -47,14 +55,32 @@ func init() {
 }
 
 func main() {
+	// metricsAddr is used by Prometheus to gather the NFD Operator resource usage data.
+	// The bind address tells Prometheus which port to scrape this data's metrics from.
 	var metricsAddr string
+
+        // enableLeaderElection should be set to 'disable' by default If we enable leader
+        // election, then only one node can run the controller manager and we will not
+        // have NFD Operator running on all nodes.
 	var enableLeaderElection bool
+
+        // probeAddr is responsible for the health probe bind address, where the health
+        // probe is responsible for determining liveness, readiness, and configuration 
+        // of the operator pods.
 	var probeAddr string
+
+	// The following 3 lines setup the CLI arguments that are used upon initilization of
+	// the operator
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+
+	// opts is created using zap to set the operator's logging to "development mode".
+	// This mode makes DPanic-level logs panic instead of just logging error events as
+	// errors. The settings are then bound to the CLI flag args and the flag args are
+	// then parsed.
```

Comment:
Maybe the description about development mode could be moved one line lower (above `Development: true`) for even tighter coupling with the context.

Reply:
Moved this one line lower like you suggested.